web/modules/custom/asset_release/asset_release.module

```diff
@@ -91,3 +94,84 @@ function asset_release_theme($existing, $type, $theme, $path) {
     ],
   ];
 }
+
+/**
+ * Implements hook_ENTITY_TYPE_access().
+ *
+ * Interrupts the permissions for updating a release. The permission is not
+ * granted if there are no transitions allowed for the current state of the
+ * node.
+ */
+function asset_release_rdf_entity_access(EntityInterface $entity, $operation, AccountInterface $account) {
```

Comment:
Indeed! We have already discussed about this. 
The conclusion of the discussions was that since we have a lot of workflows, we cannot be sure about special needs until we have at least a lot of them implemented.
After we have these, if we are sure that we can have a generic implementation for all entities, we can convert all of them in a generic one but also we should not force it to happen as it is about very few lines of code and does not make this urgent or in need of change compared to the possible risk of special handling of an entity.